### DISCUSSION TITLE
Zigbee2mqtt update

### DISCUSSION THREAD

**Discussion by U--d--o** [Koenkk/zigbee2mqtt#11970] [General]
Hello, 

I can't find instructions to update to the latest version of zigbee2mqtt. I installed it in Windows but I don't know how to update it. I can't find any documentation!

Thanks.

**Comment by robertalexa** [comment]
cd z2m_directory
git pull
npm ci

**Reply by U--d--o** [reply]
thanks for the tip, but what do I have to do if I didn't use Git for the installation?

**Reply by robertalexa** [reply]
Well, how did you install it? :) Then i can try to advise

**Reply by U--d--o** [reply]
i followed the alternative description on zigbee2mqtt for windows

"otherwise use the green Clone or download button to download the zip archive, then extract it"

today I know that git would have been better

**Reply by robertalexa** [reply]
Oh I see :)

Start by making a copy of your current z2m folder - just in case anything goes bad, you have a backup. Download the zip again, and extract it on top of you current z2m folder. Then run `npm ci` again

Alternatively, you can set it up from the start using git instead making your life easier in future. Git clone into a new folder. Run npm ci. Then copy the configuration, coordinator_backup and database files into from your old z2m setup. This way you keep your current setup.

Git Bash is a good start in Windows:
https://gitforwindows.org/

Hope this helps

**Reply by U--d--o** [reply]
Thank you very much for the tips, I'll read up on Git then!

**Reply by robertalexa** [reply]
Literally install it, open it up. Use `cd path_to_your_location` to change directory. Then git clone :) Should be a walk in the park

**Reply by U--d--o** [reply]
Hello again,

Update from Version 1.29 -> 1.30

#: git pull 
remote: Enumerating objects: 571, done.
remote: Counting objects: 100% (331/331), done.
remote: Compressing objects: 100% (9/9), done.
remote: Total 571 (delta 322), reused 329 (delta 322), pack-reused 240
Receiving objects: 100% (571/571), 932.31 KiB | 2.38 MiB/s, done.
Resolving deltas: 100% (414/414), completed with 30 local objects.
From github.com:Koenkk/zigbee2mqtt
   dddce571..cdf62ea5  master     -> origin/master
   15ccf945..4272d9a1  dev        -> origin/dev
 * [new tag]           1.30.2     -> 1.30.2
 * [new tag]           1.29.1     -> 1.29.1
 * [new tag]           1.29.2     -> 1.29.2
 * [new tag]           1.30.0     -> 1.30.0
 * [new tag]           1.30.1     -> 1.30.1
error: Your local changes to the following files would be overwritten by merge:
        package-lock.json
Please commit your changes or stash them before you merge.
Aborting
Updating dddce571..cdf62ea5

I am a little overwhelmed to decide what I should do. Is someone so kind and gives me a tip?
thank you very much in advance

**Reply by junosuarez** [reply]
This is happening because one of the files locally got out of sync with the files in git - `package-lock.json`. In this case, to upgrade we just want the new version anyway. 

run `git reset --hard` and then `git pull`.

**Reply by U--d--o** [reply]
I have done, but unfortunately leads to the following error message

Building Zigbee2MQTT... (hash changed), finished

C:\zigbee2mqtt\lib\extension\publish.ts:22
    philips.tz.effect, // Support Hue effects for groups
               ^
TypeError: Cannot read properties of undefined (reading 'effect')
    at Object.<anonymous> (C:\zigbee2mqtt\lib\extension\publish.ts:22:16)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1155:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (C:\zigbee2mqtt\lib\controller.ts:14:1)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1155:10)

**Reply by junosuarez** [reply]
Do you have custom extensions from a previous version of zigbee2mqtt? They may need to be updated to work with the latest version.
